commands.js

```diff
@@ -251,17 +252,17 @@ var commands = exports.commands = {
 			}
 		} else {
 			if ((target === 'on' || target === 'true' || !target) || !user.can('privateroom')) {
-				room.modjoin = true;
-				this.addModCommand("" + user.name + " turned on modjoin.");
+				setting = true;
 			} else if (target in Config.groups) {
-				room.modjoin = target;
-				this.addModCommand("" + user.name + " set modjoin to " + target + ".");
+				setting = target;
 			} else {
 				this.sendReply("Unrecognized modjoin setting.");
 				return false;
 			}
+			room.modjoin = setting;
+			this.addModCommand("" + user.name + (setting === true ? " turned on modjoin" : " set modjoin to " + target) + ".");
 			if (room.chatRoomData) {
-				room.chatRoomData.modjoin = true;
+				room.chatRoomData.modjoin = setting;
```

Comment:
You can just replace `true` with `room.modjoin;` and do literally nothing else.

Is there a need for all your other changes?